src/misc.asm

```diff
@@ -72,26 +92,33 @@ hook_unpause:
     LDA !ram_seg_rt_minutes : INC : STA !ram_seg_rt_minutes
 
   .done
+    ; Replace overwritten logic to enable NMI
     JSL $80834B
     RTL
 }
 
 
 gamemode_end:
 {
+   ; overwritten logic
+if !FEATURE_PAL
+    JSL $A09179
+else
     JSL $A09169
-    %a8() : LDA $4201 : ORA #$80 : STA $4201 : %a16()
-    LDA $2137 : LDA $213D : AND #$00FF : STA !ram_lag_counter
-
-    ; If mini map is disabled, we ignore artificial lag
-    LDA $05F7 : BNE +
-
-    ; Artificial lag. 41 loops ~= 1 scanline
-    LDA !sram_artificial_lag : BEQ + : ASL #4 : TAX
-    {
-        - DEX : BNE -
-    }
-    +
+endif
+
+    ; If minimap is disabled or shown, we ignore artificial lag
+    LDA $05F7 : BNE .endlag
+    LDA !ram_minimap : BNE .endlag
```

Comment:
This LDA : BNE adds 6 CPU cycles every frame when minimap is off

Reply:
Thus overall, if looking at lag every frame when minimap is off, I think we only reduced lag by one CPU cycle.  Goal was to not change lag so I'm happy with that result.